Python/correlation/tdoa.py

```diff
@@ -0,0 +1,116 @@
+"""
+Provide functions for finding the TDOA array from correlation.
+Provide function for evaluating tdoa array.
+"""
+import numpy as np
+import math
+from scipy import signal
+
+
+def give_TDOA_array(
+        correlation_matrix,
+        lag_matrix
+):
+    """
+
+    Args:
+        correlation_matrix:
+        lag_matrix:
+
+    Returns:
+        An array containing the time difference of arrival between the hydrophones, in samples.
+    """
+    tdoa_array = []
+    for i in range(len(correlation_matrix)):
+        max_lag = lag_matrix[i][np.argmax(correlation_matrix[i])]
+        tdoa_array.append(max_lag)
+    return tdoa_array
+
+
+def calculate_distance(
+        pinger_position,
+        hydro_array_positions,
+):
+    """
+    Calculate distance between source and hydrophones.
+    This function is then used to find the theoretically correct TDOA array.
+
+    Args:
+        pinger_position:
+        hydro_array_positions:
+
+    Returns:
+        Distance between pinger and the hydrophones.
+    """
+    distance = np.empty(len(hydro_array_positions))
+    for position in hydrophone_positions:
+        distance[i] = abs(pinger_position - position)
+
+    return distance
+
+
+def compare_tdoa_array(
```

Comment:
```suggestion
def check_tdoa_array_is_valid(
```
Might make more sense that a valid check returns True/False. 

Also either capitalize tdoa or don't, but keep it consistent. My vote is for lower case.